src/main/java/Glen.java

```diff
@@ -0,0 +1,177 @@
+    private static Storage storage = new Storage("./data/tasks.txt");
+    
+    public static void main(String[] args) {
+        tasks = storage.read();
+
+        System.out.println(intro());
+        Scanner scan = new Scanner(System.in);
+        String input = scan.nextLine();
+        String low = input.toLowerCase();
+        while (!low.equals("bye")) {
+            if (low.equals("list")) {
+                System.out.println(lst());
+            } else {
+                int index = Math.max(0, low.indexOf(" "));
+                String firstWord = low.substring(0, index);
+                String end = "";
+                try {
+                    end = input.substring(++index);
+                } catch (StringIndexOutOfBoundsException e) {}
+                finally {
+                    if (firstWord.equals("mark") || firstWord.equals("unmark")) {
+                        int taskIndex = -1;
+                        try {
+                            taskIndex = Integer.valueOf(end) - 1;
+                        } catch (Exception NumberFormatException) {}
+                        if (taskIndex >= 0 && taskIndex < tasks.size()) {
+                            toggle(firstWord, taskIndex);
+                        } else {
+                            System.out.println(HORLINE + "\u2639 OOPS!!! Please select a valid item to mark/unmark.\n" + HORLINE);
+                        }   
+                    } else if (firstWord.equals("deadline")) {
+                        try {
+                            int separatorIndex = end.indexOf("/by");
+        
+                            String string1 = end.substring(0, separatorIndex).trim();
+                            String string2 = end.substring(separatorIndex + 3).trim();
+
+                            System.out.println(addDeadline(string1, string2));
+                        } catch (Exception StringIndexOutOfBoundsException) {
+                            System.out.println(HORLINE + "\u2639 OOPS!!! Your request needs to be formatted as:\ndeadline <task name> /by <time>\n" + HORLINE);
+                        }
+                    } else if (firstWord.equals("event")) {
+                        try {
+                            int fromIndex = end.indexOf("/from");
+                            int toIndex = end.indexOf("/to");
+            
+                            String string1 = end.substring(0, fromIndex).trim();
+                            String string2 = end.substring(fromIndex + 6, toIndex).trim();
+                            String string3 = end.substring(toIndex + 4).trim();
+
+                            System.out.println(addEvent(string1, string2, string3));
+                        } catch (Exception StringIndexOutOfBoundsException) {
+                            System.out.println(HORLINE + "\u2639 OOPS!!! Your request needs to be formatted as:\nevent <event name> /from <start time> /to <end time>\n" + HORLINE);
```

Comment:
With long strings, maybe it will be good if you add linebreaks for better readability?